### PR TITLE
Use default in switch with assertion

### PR DESCRIPTION
**Pull request by isaachier**
If you run `cmake -DCMAKE_BUILD_TYPE=Release` for the build, you might see this warning:

```
h3/src/h3lib/lib/faceijk.c:604:13: warning: ‘edge1’ may be used uninitialized in this function [-Wmaybe-uninitialized]
             _v2dIntersect(&orig2d0, &orig2d1, edge0, edge1, &inter);
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

There are a few more of the same type. I tracked down the root cause is the switch statements. In C, an enum isn't guaranteed to be the only values passed in. For example:

```
typedef enum my_enum { x = 1, y = 2, z = 3 } my_enum;
void func(my_enum e) {
    switch (e) {
    case x:
        /* ... */
    case y:
        /* ... */
    case z:
        /* ... */
    }
    /* Do stuff */
}
/* In some code later on. */
func((my_enum) 5);
```

In the case above, `5` is not handled correctly in the switch. So I've submitted this pull request to modify the switch case like so.

```
void func(my_enum e) {
    switch (e) {
    case x:
        /* ... */
    case y:
        /* ... */
    default:
        assert(e == z);
        /* ... */
    }
    /* Do stuff */
}
```

I've argued for this position [elsewhere](https://github.com/census-instrumentation/opencensus-cpp/issues/79#issuecomment-370602770), with some disagreement, so I understand if you do not accept this PR.

### CODE REVIEW

File: src/h3lib/lib/faceijk.c [hunk 1]
```diff
@@ -593,7 +594,8 @@ void _faceIjkPentToGeoBoundary(const FaceIJK* h, int res, GeoBoundary* g) {
                     edge0 = &v1;
                     edge1 = &v2;
                     break;
-                case KI:
+                default:
```
[the remark this answers is not in the document]
Comment (isaachier) [reply src/h3lib/lib/faceijk.c]: You mean `assert`? It's a macro that only compiles in debug mode so it already avoids that penalty in releases. Look up the relationship between `assert` and `NDEBUG` for more info.

[the remark this answers is not in the document]
Comment (isaachier) [reply src/h3lib/lib/faceijk.c]: Also, I used to do something like that. I'd use `default: assert(false);`. But I found that looked pretty ugly.

Comment (isaacbrodsky) [inline src/h3lib/lib/faceijk.c]: Any thoughts on having:
```
case KI:
default:
    assert(... = KI); // Ensure we did not get here via the default label.
```

The idea here is that it uses the normal, expected idiom to go to KI, maintains the assertion, and doesn't require adding uncoverable lines.

Comment (isaachier) [reply src/h3lib/lib/faceijk.c]: OK. I mentioned above that the Google developers working on opencensus disagree with me entirely that there needs to be a default. There is a clang flag for it.

**Comment by isaacbrodsky** [comment]
@nrabinowitz I do like that suggestion, but it either causes coverage to decrease or requires LCOV_EXCL_LINE.

**Pull request merged**